packages/ui/src/validators/hooks/useStakingStatistics.tsx

```diff
@@ -0,0 +1,69 @@
+import { BN } from '@polkadot/util'
+import { useMemo } from 'react'
+import { combineLatest, map } from 'rxjs'
+
+import { useApi } from '@/api/hooks/useApi'
+import { useObservable } from '@/common/hooks/useObservable'
+
+import { ERA_DURATION } from '../constants/constant'
+
+export const useStakingStatistics = () => {
+  const { api } = useApi()
+  const activeEra = useObservable(
+    () =>
+      api?.query.staking.activeEra().pipe(
+        map((activeEra) => ({
+          eraIndex: activeEra?.unwrap().index,
+          eraStartedOn: activeEra?.unwrap().start,
+        }))
+      ),
+    [api?.isConnected]
+  )
+  const { eraIndex, eraStartedOn } = useMemo(
+    () => activeEra ?? { eraIndex: new BN(0), eraStartedOn: 0 },
+    [activeEra, api?.isConnected]
+  )
```

Comment:
This default `new eraIndex: new BN(0)` causes some extra query to be send to the rpc node (which we should avoid as much as possible). I don't think there should be any default for `activeEra`.

Instead `currentStaking`, `stakers`, `lastValidatorRewards` and `eraRewardPoints` should be `undefined` until `activeEra` is defined. Which is fine since already theses are `undefined` until the `api` connects.

E.g:
```ts
const currentStaking = useObservable(() => activeEra && api?.query.staking.erasTotalStake(activeEra.index), [activeEra, api?.isConnected])
```

Reply:
I fixed the code according to your instructions.